### ISSUE TITLE
Add rule S5876 for JS/TS

### ISSUE THREAD

**Issue by hendrik-buchwald-sonarsource** [SonarSource/SonarJS#2405]
## Rule S5876: A new session should be created during user authentication

[S5876](https://jira.sonarsource.com/browse/RSPEC-5876), part of [MMF-2154](https://jira.sonarsource.com/browse/MMF-2154).

### Summary

This rule detects missing regeneration of session ids for Passport.js. It should raise an issue if the success callback of `passport.authenticate` does not call `session.regenerate`.

### Code Samples

https://github.com/SonarSource/security-expected-issues/tree/master/javascript/rules/hotspots/RSPEC-5876

#### Noncompliant Code Example
```
app.post('/login', 
  passport.authenticate('local', { failureRedirect: '/login' }),
  function(req, res) {
    // Sensitive - no session.regenerate after login
    res.redirect('/');
  });
```

#### Compliant Solution
```
app.post('/login', 
  passport.authenticate('local', { failureRedirect: '/login' }),
  function(req, res) {
    let prevSession = req.session;
    req.session.regenerate((err) => {  // Compliant
      Object.assign(req.session, prevSession);
      res.redirect('/');
    });
  });
```
